src/main/java/core/basesyntax/strategy/BalanceStrategyHandlerImpl.java

```diff
@@ -1,20 +1,19 @@
 package core.basesyntax.strategy;
 
-import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoTransaction;
+import core.basesyntax.model.FruitTransaction;
 
 public class BalanceStrategyHandlerImpl implements StrategyHandler {
-    private static final int FRUIT_AMOUNT_POSITION_IN_ARR = 2;
-    private static final String SEPARATOR = ",";
-    private final FruitDao fruitDao;
+    private final FruitDaoTransaction fruitDao;
 
-    public BalanceStrategyHandlerImpl(FruitDao fruitDao) {
+    public BalanceStrategyHandlerImpl(FruitDaoTransaction fruitDao) {
         this.fruitDao = fruitDao;
     }
 
     @Override
-    public int doStrategy(String line) {
-        String[] lineSplit = line.split(SEPARATOR);
-        return fruitDao.getFruit(lineSplit[FRUIT_AMOUNT_POSITION_IN_ARR]).getAmount()
-                + Integer.parseInt(lineSplit[FRUIT_AMOUNT_POSITION_IN_ARR]);
+    public int doStrategy(FruitTransaction fruitTransaction) {
```

Comment:
doStrategy is a bad naming here. let's better name is 'handle'

Reply:
doStrategy is a bad naming here. let's better name is 'handle' - not fixed